src/JwplatformClient.php

```diff
@@ -0,0 +1,327 @@
+<?php
+
+namespace Jwplayer;
+
+    use Exception;
+
+class JwplatformClient {
+
+    /** @var string API client version */
+    private $_version = '2.0.0';
+
+    /** @var string API hostname */
+    private $_host = "api.jwplayer.com";
+
+    /** @var string JWPlayer API key secret */
+    private $_secret;
+
+    /**
+     * JwplatformClient constructor.
+     *
+     * @param string $secret
+     * @param string $host
+     */
+    public function __construct($secret = '', $host = '') {
+        $this->_secret = $secret;
+        if ($host) {
+            $this->_host = $host;
+        }
+
+        $this->analytics = new _AnalyticsClient($this);
+        $this->Import = new _ImportClient($this);
+        $this->Channel = new _ChannelClient($this);
+        $this->Media = new _MediaClient($this);
+        $this->WebhookClient = new _WebhookClient($this);
+        $this->advertising = new _AdvertisingClient($this);
+    }
+
+    /**
+     * Make an API request without modifications
+     *
+     * @param string $method HTTP request method
+     * @param string $url HTTP request URL
+     * @param string $body HTTP request body
+     * @param array $headers HTTP request headers
+     * @return mixed
+     */
+    public function raw_request($method, $url, $body = null, $headers = []) {
+        $url = 'https://' . $this->_host . $url;
+
+        $curl = curl_init();
+        curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);
+        curl_setopt($curl, CURLOPT_CUSTOMREQUEST, $method);
+        if ($body !== null) {
+            curl_setopt($curl, CURLOPT_POSTFIELDS, $body);
+        }
+        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($curl, CURLOPT_URL, $url);
+        $response = curl_exec($curl);
+        curl_close($curl);
+
+        $unserialized_response = @unserialize($response);
+
+        return $unserialized_response ? $unserialized_response : $response;
+    }
+
+    /**
+     * Make an API request with some convenient defaults
+     *
+     * @param string $method HTTP request method
+     * @param string $path Resource or endpoint to request
+     * @param array $body HTTP request body which will be JSON encoded
+     * @param array $headers Additional HTTP request headers
+     * @param array $query_params Additional query parameters to add to the URI
+     * @return mixed
+     */
+    public function request($method, $path, $body = null, $headers = [], $query_params = null) {
+        $headers[] = "User-Agent: ";
```

Comment:
Should this have a default?

Reply:
Yep I meant to add one here but forgot it.